### ISSUE TITLE
Regarding the cosine similarity as the loss alternative for the attack

### ISSUE THREAD

**Issue by zjysteven** [Kim-Minseon/RoCL#3]
Hi,

Thanks for releasing the code! I have a minor question on using cosine similarity as the loss alternative for the attack.
https://github.com/Kim-Minseon/RoCL/blob/440a2e80a6aeb3b9e7bd8e3332c7b566b03f5b5a/src/attack_lib.py#L151-L152

I assume in this case we should minimize the cosine similarity between the features of `x` and `target` to create the attack, right? But without taking the negative, wouldn't this attack actually maximize the cosine similarity? (And similarly, using cosine similarity as the training loss would actually minimize it?) Am I missing anything here?

This is not something directly related to RoCL, but I would appreciate it if the authors can give some confirmation on this.

Thanks!

**Comment by Kim-Minseon** [comment]
Hi,

Thank you for your comments. 
In that case, we should minimize the cosine similarity between the features of x  and target to create the attack. (And similarity, using cosine similarity as a regularizer (which is not training loss) should maximize the cosine similarity) Thank you for your comments, we found that there is a missing negative in the cosine similarity in this code. We will revise the code. I think that is why cosine similarity ablation shows 9.30 performance in robustness. 

Thank you for the report!

**Comment by zjysteven** [comment]
OK thanks for the clarification!
